### PR TITLE
OAuth section import fixes

### PR DESCRIPTION
**Pull request by joshlory**
- Allow re-importing deleted OAuth sections
- Fix bug when importing Google Classroom sections with zero students
- Implement `set_exact_student_list` to add & remove students to section

+cc @poorvasingal

### CODE REVIEW

File: dashboard/app/models/sections/omni_auth_section.rb [hunk 1]
```diff
@@ -27,18 +27,33 @@
...
-    oauth_section = where(code: code).first_or_create! do |section|
+    oauth_section = with_deleted.where(code: code).first_or_create! do |section|
       section.name = 'New Section'
       section.user_id = owner_id
       section.login_type = type
     end
-    students.each do |student|
-      oauth_section.add_student User.from_omniauth(student, {'user_type' => User::TYPE_STUDENT})
+
+    oauth_section.restore if oauth_section.deleted?
```
Comment (ashercodeorg) [inline dashboard/app/models/sections/omni_auth_section.rb:36]: Should this be `oauth_section.restore(recursive: true, recovery_window: 5.minutes) if ...`?

File: dashboard/test/models/sections/omni_auth_section_test.rb [hunk 2]
```diff
@@ -51,4 +52,24 @@ class OmniAuthSectionTest < ActiveSupport::TestCase
       assert_equal section.id, section_3.id
     end
   end
+
+  test 'set exact student list' do
+    teacher = create :teacher
+    section = create :section, user: teacher, login_type: 'clever'
+
+    students = (0...5).map do
```
Comment (ashercodeorg) [inline dashboard/test/models/sections/omni_auth_section_test.rb:60]: This could be shortened to `students = create_list :student, 5`. Also below.

Comment (joshlory) [reply dashboard/test/models/sections/omni_auth_section_test.rb:60]: TIL!

File: dashboard/app/models/sections/omni_auth_section.rb (hunk 1 above)
Comment (joshlory) [reply dashboard/app/models/sections/omni_auth_section.rb:36]: What does the `recovery_window` do?  The spec is for users to be able to re-import an OAuth classroom section, even if they've previously deleted one with the same code.

Comment (ashercodeorg) [reply dashboard/app/models/sections/omni_auth_section.rb:36]: Deleting (e.g., soft-deleting) a section causes all followers to be (soft) deleted. Doing `restore` recovers only the section. Doing `restore(recursive: true, recovery_window: 5.minutes)` causes all followers deleted within five minutes of the section delete to also be restored.

**Pull request merged**